libs/Annotator.py

```diff
@@ -1,59 +1,136 @@
-from libs import converter
+from libs.services.CIR import CIR
+from libs.services.CTS import CTS
+from libs.services.NLM import NLM
 
 
 class Annotator:
     def __init__(self, annotations):
         self.annotations = annotations
+        self.CTS = CTS()
+        self.CIR = CIR()
+        self.NLM = NLM()
 
     def add_possible_annotations(self, metadata):
         """
         Adds additional annotations to given dictionary containing metadata
 
         The method goes through specified annotations and tries to invoke
-        a particular method able to add required annotation.
+        a particular method able to add required annotation (keyword is skipped
+        if such method does not exist).
+
+        Since some metadata might depend on other metadata which might be added
+        in the process, the method iterates over the annotations until a fixpoint
+        is reached.
 
         :param metadata: given spectra metadata
         :return: annotated dictionary
         """
-        for annotation in self.annotations:
-            if annotation not in metadata:
-                try:
-                    result = getattr(self, "add_" + annotation)(metadata)
-                    if result:
-                        metadata[annotation] = result
-                        print(result)
-                except AttributeError:
-                    pass
+        added_metadata = True
+        while added_metadata:
+            added_metadata = False
+            for annotation in self.annotations:
+                if annotation not in metadata:
+                    # make sure an add method for this annotation exists
+                    try:
+                        result = getattr(self, "add_" + annotation)(metadata)
+                        if result:
+                            metadata[annotation] = result
+                            added_metadata = True
+                    except AttributeError:
+                        pass
```

Comment:
This code could be packed into a single small function, that would make it a bit better readable, but that is a detail.